common/app/shoppingcart/CartSessionUtils.java

```diff
@@ -11,21 +16,22 @@
     private CartSessionUtils() {
     }
 
-    public static long getCartItemCount(final Session session) {
-        return Optional.ofNullable(session.get(CartSessionKeys.CART_ITEM_COUNT))
-                .map(Long::parseLong)
-                .orElse(0L);
+    public static MiniCart getMiniCart(final Session session) {
+        return Optional.ofNullable(session.get(CartSessionKeys.MINI_CART))
+                .map(miniCartAsJson -> SphereJsonUtils.readObject(miniCartAsJson, MiniCart.class))
+                .orElse(new MiniCart());
     }
 
-    public static void overwriteCartSessionData(final Cart cart, final Session session) {
-        final long itemCount = cart.getLineItems().stream().mapToLong(LineItem::getQuantity).sum();
+    public static void overwriteCartSessionData(final Cart cart, final Session session, final UserContext userContext,
+                                                final ReverseRouter reverseRouter) {
+        final JsonNode miniCartAsJson = SphereJsonUtils.toJsonNode(new MiniCart(cart, userContext, reverseRouter));
```

Comment:
That's a complex thing to solve. So far this version works well enough, I'll push solving this issue post-release: https://github.com/sphereio/sphere-sunrise/issues/254